src/Sylius/Bundle/ApiBundle/CommandHandler/Checkout/AddressOrderHandler.php

```diff
@@ -64,24 +70,24 @@ public function __invoke(AddressOrder $addressOrder): OrderInterface
         );
 
         /** @var CustomerInterface|null $customer */
-        $customer =  $order->getCustomer();
+        $customer = $this->customerRepository->findOneBy(['email' => $addressOrder->email]);
```

Comment:
i am not sure about this logic. Here we need a customer which is related to order, no customer from database, because in DB the customer can exist but he can checkout his order without login. Your suggestion makes this case impossible :(